cnd/src/btsieve/bitcoin/cache.rs

```diff
@@ -0,0 +1,62 @@
+use crate::btsieve::{BlockByHash, LatestBlock};
+use bitcoin::{hashes::sha256d::Hash, util::hash::BitcoinHash, Block};
+use derivative::Derivative;
+use futures::Future;
+use futures_core::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+};
+use lru::LruCache;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+#[derive(Derivative, Clone)]
+#[derivative(Debug)]
+pub struct Cache<C> {
+    pub connector: C,
+    #[derivative(Debug = "ignore")]
+    pub block_cache: Arc<Mutex<LruCache<Hash, Block>>>,
+}
+
+impl<C> Cache<C> {
+    pub fn new(connector: C, capacity: usize) -> Cache<C> {
+        let block_cache = Arc::new(Mutex::new(LruCache::new(capacity)));
+        Cache {
+            connector,
+            block_cache,
+        }
+    }
+}
+
+impl<C> LatestBlock for Cache<C>
+where
+    C: LatestBlock<Block = Block, BlockHash = Hash> + Clone,
```

Comment:
Personally, I don't like these imports :/

It feels like pointless information to me but we can't remove it. I'd opt for `bitcoin::Block` here so that it at least adds some value (you know that this is implementation is bitcoin without having to check the imports or the file name)

Yes we are in the _context_ of bitcoin here but I find that mental jump annoying.

Thoughts?

Reply:
Can't use the macro if we don't shorted the types.  I agree the mental jump is not nice, I feel all this type info is just crap to keep the compiler happy anyways so it doesn't bother me to shorten it.  Main reason is the macro though to reduce code duplication.